eventio/simtel/objects.py

```diff
@@ -261,6 +265,56 @@ class SimTelTrackSet(EventIOObject):
 class SimTelCentEvent(EventIOObject):
     eventio_type = 2009
 
+    def __init__(self, header, parent):
+        super().__init__(header, parent)
+
+        if header.version > 2:
+            raise IOError('Unsupported CENTEVENT Version: {}'.format(header.version))
+
+        self.global_count = self.header.id
+
+    def parse_data_field(self):
+
+        event_info = {}
+        event_info['cpu_time'] = read_time(self)
+        event_info['gps_time'] = read_time(self)
+        event_info['trigger_pattern'], = read_from('<i', self)
+        event_info['data_pattern'], = read_from('<i', self)
+
+        if self.header.version >= 1:
+            tels_trigger, = read_from('<h', self)
+            event_info['n_triggered_telescopes'] = tels_trigger
+
+            event_info['triggered_telescopes'] = np.frombuffer(
+                self.read(tels_trigger * 2), dtype='<i2',
+            )
+            event_info['trigger_times'] = np.frombuffer(
+                self.read(tels_trigger * 4), dtype='<f4',
+            )
+            tels_data, = read_from('<h', self)
+            event_info['n_telescopes_with_data'] = tels_data
+            event_info['telescopes_with_data'] = np.frombuffer(
+                self.read(tels_data * 2), dtype='<i2'
+            )
+
+        if self.header.version >= 2:
+            event_info['teltrg_type_mask'] = [
+                read_utf8_like_unsigned_int(self)
+                for _ in range(tels_trigger)
+            ]
```

Comment:
Since we *know* (? do we?) that these crazy ints here are really only 1byte long since they only represent numbers between 0..7 we could replace this with `read_array(self, 'B', tels_trigger)` I think.

Reply:
If we assume these ints are only 1 byte long .. and the assumption is wrong, we can realize this by checking if the first number we read out is `>=128` that is ... checking if it has its uppermost bit set.

If so we would seek back and try again using the slower variant.